forge-game/src/main/java/forge/game/combat/Combat.java

```diff
@@ -303,8 +303,15 @@ public final Player getDefenderPlayerByAttacker(final Card c) {
 
         // maybe attack on a controlled planeswalker?
         if (defender instanceof Card) {
-            return ((Card) defender).getController();
+            Card def = (Card)defender;
+            if (def.isBattle() && def.getType().hasSubtype("Siege")) {
+                return def.getProtectingPlayer();
```

Comment:
imo that should be for all of Battle, just Siege defines how the Protector is chosen